packages/compiler/src/init/init.ts

```diff
@@ -222,35 +246,83 @@ async function downloadTemplates(
     ]);
   }
 
-  validateTemplateDefinitions(json, file);
-  return json;
-}
-
-async function selectTemplate(
-  host: CompilerHost,
-  templatesUrl: TemplatesUrl | undefined
-): Promise<InitTemplate> {
-  const templates =
-    templatesUrl === undefined ? builtInTemplates : await downloadTemplates(host, templatesUrl);
-  return promptTemplateSelection(templates);
+  //validateTemplateDefinitions(json, file);
+  return json as Record<string, any>;
 }
 
 async function promptTemplateSelection(
-  templates: Record<string, InitTemplate>
-): Promise<InitTemplate> {
+  templates: Record<string, any>,
+  templatesUrl: TemplatesUrl | undefined
+): Promise<string> {
   const { templateName } = await prompts({
     type: "select",
     name: "templateName",
     message: "Please select a template",
     choices: Object.entries(templates).map(([id, template]) => {
-      return { value: id, description: template.description, title: template.title };
+      return {
+        value: id,
+        description: template.description,
+        title:
+          template.title +
+          `\tmin compiler ver: ${
+            template.minimalCompilerVersion ? template.minimalCompilerVersion : "-not specified-"
+          }`,
+      };
     }),
   });
+
   const template = templates[templateName];
   if (!template) {
     throw new Error(`Unexpected error: Cannot find template ${templateName}`);
   }
-  return template;
+
+  return templateName;
+}
+
+type ValidationResult = {
+  valid: boolean;
+  diagnostics: readonly Diagnostic[];
+};
+
+async function validateTemplate(template: any, templatesUrl: TemplatesUrl): Promise<boolean> {
+  // After selection, validate the template definition
+  const currentCompilerVersion = MANIFEST.version;
+  const validationTarget = templatesUrl.file as SourceFile;
+  let validationResult: ValidationResult;
+  // 1. If current version > minimalCompilerVersion, proceed with strict validation
+  if (template.minimalCompilerVersion && currentCompilerVersion > template.minimalCompilerVersion) {
```

Comment:
```suggestion
  if (template.minimalCompilerVersion && currentCompilerVersion > template.minimalCompilerVersion) {
```
not sure this will work correctly, think we should either add dependency on `semver` package to validate that for us or have a more complex comparaison

Reply:
While this generally working fine for version number not containing wildcards, I agree semver offer better capability. Will sync up tomorrow on semver version dependencies.